src/retry_manager.rs

```diff
@@ -0,0 +1,165 @@
+        Self {
+            id: Uuid::new_v4().to_string(),
+            recipient,
+            text,
+            attachment_path,
+            status: DeliveryStatus::Pending,
+            retry_count: 0,
+            created_at: Self::current_timestamp(),
+            last_attempt_at: None,
+        }
+    }
+
+    pub fn should_retry(&self, max_retries: u32, retry_delay_seconds: u64) -> bool {
+        if !matches!(self.status, DeliveryStatus::Failed(_)) {
+            return false;
+        }
+
+        if let DeliveryStatus::Failed(ref reason) = self.status
+            && reason
+                .to_lowercase()
+                .contains("websocket closing while waiting")
+        {
+            return false;
+        }
+
+        if self.retry_count >= max_retries {
+            return false;
+        }
+
+        if let Some(last_attempt) = self.last_attempt_at {
+            let now = Self::current_timestamp();
+            let delay_ms = retry_delay_seconds * 1000;
+            (now - last_attempt) >= delay_ms
+        } else {
+            true
+        }
+    }
+
+    pub fn mark_sending(&mut self) {
+        self.status = DeliveryStatus::Sending;
+        self.last_attempt_at = Some(Self::current_timestamp());
+        self.retry_count += 1;
+    }
+
+    pub fn mark_sent(&mut self) {
+        self.status = DeliveryStatus::Sent;
+    }
+
+    pub fn mark_failed(&mut self, reason: String) {
+        self.status = DeliveryStatus::Failed(reason);
+    }
+
+    fn current_timestamp() -> u64 {
+        SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap()
+            .as_millis() as u64
+    }
+}
+
+pub struct RetryManager {
+    outgoing_messages: HashMap<String, OutgoingMessage>,
+    max_retries: u32,
+    retry_delay_seconds: u64,
+}
+
+impl Default for RetryManager {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl RetryManager {
+    pub fn new() -> Self {
+        Self {
+            outgoing_messages: HashMap::new(),
+            max_retries: 3,
+            retry_delay_seconds: 30,
+        }
+    }
+
+    pub fn add_message(&mut self, message: OutgoingMessage) -> String {
+        let id = message.id.clone();
+        self.outgoing_messages.insert(id.clone(), message);
+        id
+    }
+
+    pub fn mark_sent(&mut self, message_id: &str) {
+        if let Some(msg) = self.outgoing_messages.get_mut(message_id) {
+            msg.mark_sent();
+        }
+    }
+
+    pub fn mark_sending(&mut self, message_id: &str) {
+        if let Some(msg) = self.outgoing_messages.get_mut(message_id) {
+            msg.mark_sending();
+        }
+    }
+
+    pub fn mark_failed(&mut self, message_id: &str, reason: String) {
+        if let Some(msg) = self.outgoing_messages.get_mut(message_id) {
+            msg.mark_failed(reason);
+        }
+    }
+
+    pub fn get_messages_to_retry(&mut self) -> Vec<OutgoingMessage> {
```

Comment:
```suggestion
    pub fn messages_to_retry(&mut self) -> Vec<OutgoingMessage> {
```
Don't forget to change usages.